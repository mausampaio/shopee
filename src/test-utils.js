import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

export const mockProduct = {
  id: 1,
  name: "Produto de teste",
  price: 1.0,
  description: "Produto gerado para testes",
  url: "produto-de-teste",
  image: "./produto-teste.jpg",
};

export const mockCart = {
  id: mockProduct.id,
  name: mockProduct.name,
  price: mockProduct.price,
  quantity: 10,
};

export const mockContext = {
  cart: [],
  products: [],
  product: {},
  loading: false,
  getCartTotal: () => 0,
};

const RouterProvider = ({ children }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};

const customRender = (ui, options) => {
  return render(<RouterProvider>{ui}</RouterProvider>, { ...options });
};

export * from "@testing-library/react";

export { customRender as render };
