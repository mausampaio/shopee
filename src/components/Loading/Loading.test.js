import { render } from "../../test-utils";
import "jest-styled-components";

import { Loading } from "./index";

it("should match snapshot", () => {
  const { asFragment } = render(<Loading />);
  expect(asFragment()).toMatchSnapshot();
});
