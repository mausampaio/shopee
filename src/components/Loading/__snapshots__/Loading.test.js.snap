// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  border: 8px solid #e7e7e7;
  border-radius: 50%;
  border-top: 8px solid var(--secondary);
  width: 80px;
  height: 80px;
  margin: 64px auto;
  -webkit-animation: dHAcpB 2s linear infinite;
  animation: dHAcpB 2s linear infinite;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
